tree/ntuple/test/ntuple_evolution.cxx

```diff
@@ -863,3 +863,61 @@ struct RenamedIntermediateDerived : public RenamedIntermediate2 {
       EXPECT_THAT(err.what(), testing::HasSubstr("incompatible type name for field"));
    }
 }
+
+TEST(RNTupleEvolution, StreamerField)
+{
+   FileRaii fileGuard("test_ntuple_evolution_streamer_field.root");
+   fileGuard.PreserveFile();
+
+   ExecInFork([&] {
+      // The child process writes the file and exits, but the file must be preserved to be read by the parent.
+      fileGuard.PreserveFile();
+
+      ASSERT_TRUE(gInterpreter->Declare(R"(
+struct StreamerField {
+   int fInt = 1;
+   int fAnotherInt = 3;
+
+   ClassDefNV(StreamerField, 2)
+};
+)"));
+
+      auto model = RNTupleModel::Create();
+      model->AddField(std::make_unique<ROOT::RStreamerField>("f", "StreamerField"));
+
+      auto writer = RNTupleWriter::Recreate(std::move(model), "ntpl", fileGuard.GetPath());
+      writer->Fill();
+
+      void *ptr = writer->GetModel().GetDefaultEntry().GetPtr<void>("f").get();
+      DeclarePointer("StreamerField", "ptrStreamerField", ptr);
+      ProcessLine("ptrStreamerField->fInt = 2;");
+      writer->Fill();
+
+      // Reset / close the writer and flush the file.
+      writer.reset();
+   });
+
+   ASSERT_TRUE(gInterpreter->Declare(R"(
+struct StreamerField {
+   int fInt = 0;
+   int fAdded = 137;
```

Comment:
To test the indirect collection of StreamerInfo, there should be here a pointer to another class (possibly initialized with an instance of a derived class).

Reply:
I think we already test that with https://github.com/root-project/root/blob/031f6bd8789ec363ed5552531657865b6296d2fe/tree/ntuple/test/rfield_streamer.cxx#L126

Reply:
Technically the test linked will succeed whether or not the `StreamerInfo` are stored ... since they are still in memory from the already loaded dictionary.